test/net/sourceforge/kolmafia/session/InventoryManagerTest.java

```diff
@@ -296,6 +301,34 @@ public void willNotRetrieveIfAlreadyBoughtFireworkHat() {
     }
   }
 
+  @Nested
+  class PerUserVariableItems {
+    @Test
+    public void RingWillSetModsForUser() {
+      var builder = new FakeHttpClientBuilder();
+      builder.client.addResponse(200, html("request/test_desc_item_ring_2crs.html"));
+
+      var cleanups = new Cleanups(withHttpClientBuilder(builder), withItem(ItemPool.RING, 1));
+      try (cleanups) {
+        Modifiers mods = ModifierDatabase.getModifiers(ModifierType.ITEM, ItemPool.RING);
+        assertEquals(
+            0, mods.getDouble(DoubleModifier.MONSTER_LEVEL), "Value should not be set before desc");
+
+        String descId = ItemDatabase.getDescriptionId(ItemPool.RING);
```

Comment:
How about just calling InventoryManager.checkRing()?
That will do the same thing - fetch and parse the desc_item - but you won't have to duplicate the implementation.

As a bonus, it increases coverage. :)

Reply:
yep, that works.